libkbfs/mdserver_remote.go

```diff
@@ -441,11 +440,22 @@ func (md *MDServerRemote) signalObserverLocked(observerChan chan<- error, id tlf
 // Helper used to retrieve metadata blocks from the MD server.
 func (md *MDServerRemote) get(ctx context.Context, id tlf.ID,
 	handle *tlf.Handle, bid BranchID, mStatus MergeStatus,
-	start, stop MetadataRevision) (tlf.ID, []*RootMetadataSigned, error) {
+	start, stop MetadataRevision) (tlfID tlf.ID, rmdses []*RootMetadataSigned, err error) {
 	// figure out which args to send
 	if id == tlf.NullID && handle == nil {
 		panic("nil tlf.ID and handle passed into MDServerRemote.get")
 	}
+	var reqStr string
+	if id != tlf.NullID {
+		reqStr = fmt.Sprintf("%s", id)
```

Comment:
Ideally we'd only do these `Sprintf`s if tracing is on.  You could build a simple class that only computes this once `String()`is called, and then pass that to `LazyTrace`?  Or just pass both to `LazyTrace` and let the wrong one print as the zero value?  Or call `LazyTrace` inside the `if` statements? Not a big deal, but would be nice.